src/index.ts

```diff
@@ -158,32 +159,40 @@ class CruxPayPeer extends EventEmitter {
         // Setting up the default modules as fallbacks
         this._storage =  this._options.storage || new storage.LocalStorage();
         this._encryption = this._options.encryption || encryption.Encryption;
-        this._nameservice = this._options.nameservice;
+        this._nameService = this._options.nameService;
         this.walletClientName = this._options.walletClientName;
 
         log.info(`Config mode:`, config.CONFIG_MODE);
         log.info(`CruxPayPeer Initialised`);
     }
 
     public async init() {
-        const configService = new BlockstackConfigurationService(this.walletClientName);
-        await configService.init();
-        if (!this._nameservice) {
-            this._nameservice = await configService.getBlockstackServiceForConfig();
-        }
+        let configService;
 
         if (this._hasPayIDClaimStored()) {
             const payIDClaim = this._storage.getJSON("payIDClaim");
             // log.debug(`Local payIDClaim:`, payIDClaim)
             this._setPayIDClaim(new PayIDClaim(payIDClaim as ICruxPayClaim, { getEncryptionKey: this._getEncryptionKey }));
+            const ns: blockstackService.BlockstackService = new blockstackService.BlockstackService();
+            await this.getPayIDClaim().decrypt();
+            await ns.restoreIdentity(this.getPayIDClaim().virtualAddress as string, {secrets: (this._payIDClaim as PayIDClaim).identitySecrets});
+            const status = await ns.getRegistrationStatus({secrets: this.getPayIDClaim().identitySecrets});
+            await this.getPayIDClaim().encrypt();
```

Comment:
For internal usage, please rely on `this._payIDClaim` rather than `this.getPayIDClaim()`